deepchem/molnet/load_function/material_datasets/load_mp_formation_energy.py

```diff
@@ -0,0 +1,205 @@
+  **kwargs : additional optional arguments.
+
+  Returns
+  -------
+  tasks, datasets, transformers : tuple
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  References
+  ----------
+  .. [1] A. Jain*, S.P. Ong*, et al. (*=equal contributions) The Materials Project: A materials genome approach to accelerating materials innovation APL Materials, 2013, 1(1), 011002. doi:10.1063/1.4812323 (2013).
+
+  .. [2] Dunn, A. et al. "Benchmarking Materials Property Prediction Methods: The Matbench Test Set and Automatminer Reference Algorithm." https://arxiv.org/abs/2005.00707 (2020)
+
+  Examples
+  --------
+  >> import deepchem as dc
+  >> tasks, datasets, transformers = dc.molnet.load_mp_formation_energy(reload=False)
+  >> train_dataset, val_dataset, test_dataset = datasets
+  >> n_tasks = len(tasks)
+  >> n_features = train_dataset.get_data_shape()[0]
+  >> model = dc.models.MultitaskRegressor(n_tasks, n_features)
+
+  """
+
+  # Featurize
+  logger.info("About to featurize mp formation energy dataset.")
+  my_tasks = ['formation_energy']  # machine learning targets
+
+  # Get DeepChem data directory if needed
+  if data_dir is None:
+    data_dir = DEFAULT_DIR
+  if save_dir is None:
+    save_dir = DEFAULT_DIR
+
+  if issubclass(featurizer, MaterialStructureFeaturizer):
+    featurizer = featurizer(**featurizer_kwargs)
+  else:
+    raise TypeError(
+        "featurizer must be a subclass of MaterialStructureFeaturizer.")
+
+  if issubclass(splitter, Splitter):
+    splitter = splitter()
+  else:
+    raise TypeError("splitter must be a subclass of Splitter.")
+
+  # Reload from disk
+  if reload:
+    featurizer_name = str(featurizer.__class__.__name__)
+    splitter_name = str(splitter.__class__.__name__)
+    save_folder = os.path.join(save_dir, "mp-forme-featurized", featurizer_name,
+                               splitter_name)
+
+    loaded, all_dataset, transformers = deepchem.utils.save.load_dataset_from_disk(
+        save_folder)
+    if loaded:
+      return my_tasks, all_dataset, transformers
+
+  # First type of supported featurizers
+  supported_featurizers: List[str] = [
+      'CGCNNFeaturizer',
+      'SineCoulombMatrix',
+  ]
+
+  # Load .tar.gz file
+  if featurizer.__class__.__name__ in supported_featurizers:
+    dataset_file = os.path.join(data_dir, 'mp_formation_energy.tar.gz')
+    deepchem.utils.untargz_file(dataset_file, dest_dir=data_dir)
+    dataset_file = os.path.join(data_dir, 'mp_formation_energy.json')
+
+    if not os.path.exists(dataset_file):
+      deepchem.utils.download_url(url=MPFORME_URL, dest_dir=data_dir)
+      deepchem.utils.untargz_file(
+          os.path.join(data_dir, 'mp_formation_energy.tar.gz'), data_dir)
```

Comment:
I seem these lines doesn't work... (see #2083)
Please check whether to work except pytest like this.

```bash
$ python
>>> import deepchem as dc
>>> tasks, datasets, transformers = dc.molnet.load_mp_formation_energy(reload=False)
```